File: app/src/main/java/com/capstone/bankit/ui/main/ui/profile/ProfileFragment.java
package com.capstone.bankit.ui.main.ui.profile;

import android.content.Intent;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProvider;

import com.capstone.bankit.databinding.FragmentProfileBinding;
import com.capstone.bankit.ui.editprofile.EditProfileActivity;


public class ProfileFragment extends Fragment {

    private FragmentProfileBinding binding;

    public View onCreateView(@NonNull LayoutInflater inflater,
                             ViewGroup container, Bundle savedInstanceState) {
        binding = FragmentProfileBinding.inflate(inflater, container, false);

        setListeners();

        return binding.getRoot();
    }

    private void setListeners() {
        binding.btnEdit.setOnClickListener(v -> {
            startActivity(new Intent(getActivity(), EditProfileActivity.class));
        });
    }
}

File: app/src/main/java/com/capstone/bankit/ui/main/profile/ProfileFragment.kt
package com.capstone.bankit.ui.main.profile

import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.capstone.bankit.databinding.FragmentProfileBinding
import com.capstone.bankit.ui.auth.AuthActivity
import com.capstone.bankit.ui.editprofile.EditProfileActivity
import com.capstone.bankit.utils.TokenManager

class ProfileFragment : Fragment() {
    private lateinit var binding: FragmentProfileBinding

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?, savedInstanceState: Bundle?,
    ): View {
        binding = FragmentProfileBinding.inflate(inflater, container, false)

        setListeners()
        setData()

        return binding.root
    }

    private fun setData() {
        val name = TokenManager.getInstance(requireContext())?.name
        val email = TokenManager.getInstance(requireContext())?.email

        binding.edEmail.setText(email)
        binding.edUsername.setText(name)
        binding.tvUsername.text = name
    }

    private fun setListeners() {
        binding.edEmail.isEnabled = false
        binding.edUsername.isEnabled = false
        binding.edPassword.isEnabled = false

        binding.btnEdit.setOnClickListener {
            startActivity(
                Intent(
                    activity,
                    EditProfileActivity::class.java
                )
            )
        }
        binding.btnLogout.setOnClickListener {
            TokenManager.getInstance(requireContext())?.clearToken()
            startActivity(Intent(
                activity, AuthActivity::class.java
            ))
            activity?.finish()
        }
    }
}